contracts/AJMarket.sol

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+
+/// @custom:security-contact security@ajcoin.xyz
+contract AJMarket is Ownable {
+    using Counters for Counters.Counter;
+
+    IERC20 ajCoin;
+
+    // Total count of all listings
+    Counters.Counter private _listingCount;
+
+    // Mapping from listing ID to supply
+    mapping(uint256 => uint256) private _listingSupply;
+
+    // Mapping from listing ID to price
+    mapping(uint256 => uint256) private _listingPrice;
+
+    // Total count of all orders
+    Counters.Counter private _orderCount;
+
+    // Mapping from order ID to listing ID
+    mapping(uint256 => uint256) private _orderListing;
+
+    // Mapping from order ID to user address
+    mapping(uint256 => address) private _orderUser;
+
+    // Mapping from order ID to price
+    mapping(uint256 => uint256) private _orderPrice;
+
+    // Mapping user address to order count
+    mapping(address => uint256) private _orderCountForUser;
+
+    // Mapping from user to list of order IDs
+    mapping(address => mapping(uint256 => uint256)) private _orders;
+
```

Comment:
Would there be a mapping for orderStatus?

Reply:
And also a view call to get the status of all of their orders, possibly even a call to get all orders of a specific status.  Maybe even record the block when each order status change was submitted.  All of these suggestions are extra but ways I interact with any service that handles online orders (Shopify sites, Amazon etc).

Reply:
I wanted to keep that off chain because there's already an element of trust on me to say I actually shipped it, so I figured there's limited value in that part being on chain.

Reply:
that's also why I made the call to "index" orders by users as I expect to need to fetch them from the UI for a specific user